api/lcp/encrypt.py

```diff
@@ -4,49 +4,56 @@
 import re
 import subprocess
 from json import JSONEncoder
+from typing import Optional
 
 from flask_babel import lazy_gettext as _
 
 from api.lcp import utils
 from core.exceptions import BaseError
-from core.model.configuration import (
-    ConfigurationAttributeType,
-    ConfigurationGrouping,
-    ConfigurationMetadata,
+from core.integration.settings import (
+    BaseSettings,
+    ConfigurationFormItem,
+    ConfigurationFormItemType,
+    FormField,
 )
+from core.model.integration import IntegrationConfiguration
 
 
 class LCPEncryptionException(BaseError):
     """Raised in the case of any errors occurring during LCP encryption process"""
 
 
-class LCPEncryptionConfiguration(ConfigurationGrouping):
-    """Contains different settings required by LCPEncryptor"""
-
+class LCPEncryptionConstants:
     DEFAULT_LCPENCRYPT_LOCATION = "/go/bin/lcpencrypt"
     DEFAULT_LCPENCRYPT_DOCKER_IMAGE = "readium/lcpencrypt"
 
-    lcpencrypt_location = ConfigurationMetadata(
-        key="lcpencrypt_location",
-        label=_("lcpencrypt's location"),
-        description=_(
-            "Full path to the local lcpencrypt binary. "
-            "The default value is {}".format(DEFAULT_LCPENCRYPT_LOCATION)
+
+class LCPEncryptionSettings(BaseSettings):
+    lcpencrypt_location: Optional[str] = FormField(
```

Comment:
Should this be optional? It has a default, so we probably don't want it to be optional, and instead want to error if there is no data in the field.